src/device/ping-time.h

```diff
@@ -0,0 +1,6 @@
+#pragma once
+
+namespace PingTime {
```

Comment:
This is not ping related, the name should be TimeUtils or something.

Reply:
It is ping related: it is the abstract system time interface that the PingDevice class depends on. It is designed to use with PingDevice.

Reply:
It's time and not ping related, ping uses it to do stuff and not the other way around. It does not deals with ping related protocol, or ping sensor functionality, or anything related to ping. The functions inside this class are only time and the information that is provided is also time, time is the only thing that this names use and provides, so Ping as a prefix does not make sense IMO. It should be TimeUtils, TimeHelper, or something that indicates that such functions wraps std::chrono to help with time functionalities. 

To exemplify, I can use such functions to help with network information of github, flasher response time, calculate sound speed or time elapsed inside a function. Such things are not ping related and not necessary something that is ping explicit, so the namespace and file name are not clear.

Reply:
This is an interface designed only for PingDevice. That is the only purpose it is meant to serve. Yes, you are right that many other applications in program require time management, and the code is similar on all of them. This is the standard interface that PingDevice uses.

It is ping related.